src/index.ts

```diff
@@ -54,11 +54,13 @@ async function checkVersion(
   const outdated: Record<string, Version> = {};
   if (upstreamVersion !== null) {
     for (const [key, version] of rest) {
-      if (version.main !== upstreamVersion.version.main) {
+      if (!upstreamVersion.version.equivalent(version)) {
         if (upstreamVersion.versions !== undefined) {
-          const fullVersion = upstreamVersion.versions.find(({ main }) => {
-            return main === version.main;
-          });
+          const fullVersion = upstreamVersion.versions.find(
+            (upstreamVersion) => {
+              return upstreamVersion.equivalent(version);
+            },
+          );
```

Comment:
Versions from `flake.lock` only contain commit hashes but this isn't particularly useful when displayed as we don't know how far behind it is. So we essentially check if the commit hash matches that of a nixpkgs version we have listed and substitute it so that we get a nicer human-readable output.